### ISSUE TITLE
Some migration issues remaining 

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#670]
Following are issues remaining based on comparing alpha.hydroshare.org and www.hydroshare.org at this time.

1. Contributors may be repeated.  For example in https://alpha.hydroshare.org/resource/12d195906f2c41918cb24e11a5c3ab60/ Jon Goodall is repeated as a contributor, but appears only once in https://www.hydroshare.org/resource/12d195906f2c41918cb24e11a5c3ab60/

2.  The resource https://alpha.hydroshare.org/resource/2cddae40e9594c21b947fdbbe4225439/ lists HydroShare (admin) as the first author and first owner.  This is not right.

3.  Band information method is not properly transferred for raster resources.  Specifically https://alpha.hydroshare.org/resource/877bf9ed9e66468cadddb229838a9ced/ does not replicate https://www.hydroshare.org/resource/877bf9ed9e66468cadddb229838a9ced/.  The new resource only holds the end of the full method in the www resource.  This appears in the Resource specific tab in the interface and the resourcemetadata.xml file.

**Comment by selimnairb** [comment]
@dtarb  For the [duplicate creators](https://alpha.hydroshare.org/resource/12d195906f2c41918cb24e11a5c3ab60/), I'm not seeing this in my test environment, but I know why this is happening during our migration dry run.  Working on the best way to handle this.  The problem is this resource depends on another, so we have to re-create some of the metadata after the resource it depends on is created, leading to some unintentional duplication.

**Comment by hyi** [comment]
@dtarb @selimnairb My updated LoadAccessControl script should be able to take care of No. 2 in the list since if admin is not in the owner list according to the old access control info, this admin will be removed from the owner/creator list. We will make sure this is addressed in another round of migration dry-run.

**Comment by ironfroggy** [comment]
If this is happening with only a few resources and only a few issues, is it a more effective use of time to move forward and fix those issues by hand afterwards?

**Comment by selimnairb** [comment]
@ironfroggy Agreed that we are rapidly reaching diminishing returns.  Unless there is another problem found affecting nearly all resources, this should be the last round of iteration on this (in my opinion).  We should correct the remaining problems by hand and move on.

**Comment by dtarb** [comment]
Fair enough.  My checks were spot checks though so finding all the problems is hard, and there is always the probability that these issues are indicative of some more systemic problem that has been their cause that should be understood (and prioritized for being addressed based on its implications).

**Comment by dtarb** [comment]
Tests on beta.hydroshare.org following new deploy by @mjstealey show that 1 and 3 appear to be fixed but 2 still remains as a problem.

The following resource has HydroShare (Admin) as first author.
https://beta.hydroshare.org/resource/2cddae40e9594c21b947fdbbe4225439/

This is also a problem for
https://beta.hydroshare.org/resource/50fa9c5b3b5b4263bf4f1752a8b6723c/

and for
https://beta.hydroshare.org/resource/f68ade31c1cf4b2e82130520f4cc7ea6/
The last is a private resource owned by Tian.

This appears to be a systemic problem where the first author is not a HydroShare user.  Authors (dublin core creators) should be entered from the bag and the bag only.

**Comment by selimnairb** [comment]
@dtarb If this is the only outstanding problem, I believe we should fix it by hand.

**Comment by hyi** [comment]
@dtarb My updated script only removes admin from owner list by calling unshare_resource_with_user(), but did not handle author list, which explains why admin remains in author list although it is removed from owner list if you compare beta and alpha. @pkdash is there a simple function to call to remove a user from author list? If there is, I can add that call into my script to remove it from author list also. Otherwise, we will have to go to edit it out by hand unfortunately.

**Comment by selimnairb** [comment]
@hyi I can pretty easily exclude the admin user (user id 1) from being added to contributors metadata, would this do the trick?

**Comment by hyi** [comment]
@selimnairb I am not sure, have to look into the code, but it certainly helps if not work completely.

**Comment by hyi** [comment]
@selimnairb @pkdash Looks like the admin user has to be removed from both contributors and creators metadata, remove() method of the Party class might do the trick, but @pkdash will have to advise on what is the best way to remove admin from both contributors and creators metadata.

**Comment by selimnairb** [comment]
@hyi By doing this, I believe we would still need to remove admin as owner by hand, but admin would no longer appear as an author.

**Comment by hyi** [comment]
@selimnairb My script will take care of removing admin from owner list, but we need to add method either in my post-processing load access control script or your bag ingestion code to remove admin from creators and contributors metadata after the resource is initially created.

**Comment by dtarb** [comment]
I am OK if we remove these by hand, but I do not understand why admin needs to be added to authors in the first place.  It should just take the authors from the xml in github.  This may reflect a misunderstanding of how author (dublin core creator) information should really be added.  For the API and all interfaces when creating a resource, if the user provides author information that should take precedence.  Only if the user does not provide author information does the system automatically fill it in with the user information.

**Comment by hyi** [comment]
@dtarb @selimnairb @horsburgh @pkdash I believe the reason admin is used to initially create these few resources is that all creators stored in resourcemetadata.xml file are not Hydroshare users, and we don't store resource owners (which are part of access control) in the xml file currently, so @selimnairb's bag ingestion code does not have any valid hydroshare user to use to initially create the resource which is why Brian just picked admin in this special case.

**Comment by dtarb** [comment]
I worry that I am being repetitive, but we have to get straight in our data model that authors = dublin core creators are different from owners.  The code initially entering a resource can use admin as the owner, but should use the creators from the XML who do not have to be hydroshare users.

**Comment by selimnairb** [comment]
@dtarb Yes, we do need to get authors and creators straight.  As we don't currently store owner in the resourcemap or resourcemetadata, for resources whose authors are not HS users, there is no way to create a resource from bag ingestion unless we assign admin as the owner.  What I will do is exclude userID 1 from being included as an author in the resource metadata.

**Comment by dtarb** [comment]
The owner problem is actually solved on beta.  It is the author problem that remains.

**Comment by hyi** [comment]
@dtarb I have updated my post-processing script to remove admin from author and contributor lists as well as change resource creator and last_changed_by attributes to be the first valid owner according to the old access control info. I tested the updated script on beta and it appears to work for those troublesome resources you pointed out above with admin removed properly. Hopefully this updated script is good enough for the final migration without requiring us to do any edit by hand.
